### PR TITLE
MSPSDS-869 notification banner styles

### PR DESCRIPTION
**Pull request by MatthewBurstein**
@broder @jasiekmiko it would be good to get this reviewed and merged as soon as possible because my next ticket needs the component :) 

## Checklist:

- [X] Automated checks are passing locally.
### General testing
- [X] Test without javascript
- [X] Test on small screen
### Accessibility testing
- [X] Tested with one screen reader
- [X] Zoom page to 400% - content still visible
- [ ] Disable css - does content make sense and appear in a logical order?
- [ ] Passes automated checker (automated in build or manual)

### CODE REVIEW

File: shared-web/app/views/components/_banner.html.slim [hunk 1]
```diff
@@ -0,0 +1,26 @@
+ruby:
+  banner_classes = "banner"
+  if local_assigns[:type] && type != :information
```
Comment (broder) [inline shared-web/app/views/components/_banner.html.slim]: Do we need to do `type.to_sym` here? Or can we remove `type.to_sym` from some of the conditionals below?

File: shared-web/app/views/components/_banner.html.slim [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  elsif local_assigns[:classes]
+    banner_classes += " #{classes}"
+  end
+  content = capture do
+    local_assigns[:html] || local_assigns[:text]
+  end
+
+div class=banner_classes
+  .banner__message
+    -if local_assigns[:icon_fallback_test]
```
Comment (broder) [inline shared-web/app/views/components/_banner.html.slim]:
```suggestion
    -if local_assigns[:iconFallbackText]
```

File: shared-web/app/views/components/_banner.html.slim [hunk 3]
```diff
@@ -0,0 +1,26 @@
+ruby:
```
Comment (broder) [inline shared-web/app/views/components/_banner.html.slim]: Should we call this file `_hmcts_banner` for consistency with the other components?

File: shared-web/app/views/components/_banner.html.slim [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    banner_classes += " #{classes}"
+  end
+  content = capture do
+    local_assigns[:html] || local_assigns[:text]
+  end
+
+div class=banner_classes
+  .banner__message
+    -if local_assigns[:icon_fallback_test]
+      span.hmcts-banner__assistive = local_assigns[:iconFallbackText]
```
Comment (broder) [inline shared-web/app/views/components/_banner.html.slim]:
```suggestion
      span.banner__assistive = local_assigns[:iconFallbackText]
```

File: shared-web/app/views/components/_banner.html.slim [hunk 5]
```diff
@@ -0,0 +1,26 @@
+ruby:
+  banner_classes = "banner"
+  if local_assigns[:type] && type != :information
+    banner_classes += " banner--#{type.to_s}"
+  elsif local_assigns[:classes]
```
Comment (jasiekmiko) [inline shared-web/app/views/components/_banner.html.slim]: I'm not sure this wants to be an `elseif` - I can see situations where I want to use both `type` and `classes` params. Is there anything that would make them exclusive?

File: shared-web/app/views/components/_banner.html.slim [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    banner_classes += " #{classes}"
+  end
+  content = capture do
```
Comment (broder) [inline shared-web/app/views/components/_banner.html.slim]: Do we need to capture this or can we just inline it below?

File: shared-web/app/views/components/_banner.html.slim [hunk 7]
```diff
@@ -0,0 +1,26 @@
+ruby:
+  banner_classes = "banner"
```
Comment (jasiekmiko) [inline shared-web/app/views/components/_banner.html.slim]: if you make it a singleton array that you push extra classes onto, then you don't need the awkward leading whitespaces below. slim natively accepts arrays for attributes, too!

File: shared-web/app/views/components/_banner.html.slim (hunk 3 above)
Comment (jasiekmiko) [inline shared-web/app/views/components/_banner.html.slim]: Was this adapted from `https://github.com/hmcts/frontend/blob/master/src/components/banner/template.njk` like the css? Would be good documenting it.

Also, govuk components can rely on the design system to document the component interface. Do we have an equivalent for this? If not, we should document it ourselves here.

File: shared-web/app/views/components/_banner.html.slim (hunk 2 above)
Comment (jasiekmiko) [inline shared-web/app/views/components/_banner.html.slim]: I think this should say "text", not "test"?

File: shared-web/app/views/components_gallery/banner.html.slim [hunk 8]
```diff
@@ -0,0 +1,32 @@
+h1.govuk-heading-xl Breadcrumbs component
```
Comment (jasiekmiko) [inline shared-web/app/views/components_gallery/banner.html.slim]: Nice! Can we add a fallback text example, too?

File: shared-web/app/views/components_gallery/banner.html.slim [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (broder) [inline shared-web/app/views/components_gallery/banner.html.slim]: Banner component?

Comment (MatthewBurstein) [reply shared-web/app/views/components_gallery/banner.html.slim]: 🤦‍♀️

**Pull request merged**